### PR TITLE
Prepare for Unix current culture changes

### PR DESCRIPTION
**Pull request by ellismg**
We are going to change the default locale we use on Unix when we can't
figure out one based on LANG and friends to be Invariant instead of
en-US-POSIX, since that provides a better default (the collation rules
in en-US-POSIX are very strange).

Two tests depend on the current behavior, one that explicitly tests the
default (which I will update after the coresponding CoreCLR changes have
landed) and one which does not work currently if the current culture is
Invariant (I found this via directed testing where I ran all the tests
with empty LANG vars).

Fix one to run out of process and just set the CurrentCulture to
en-US (it should not impact what the tests wants to cover) and remove
the other test.

### CODE REVIEW

File: src/System.Globalization/tests/RegionInfo/RegionInfoTests.Properties.cs [hunk 1]
```diff
@@ -2,19 +2,27 @@
...
+            RemoteInvoke(() =>
+            {
+                CultureInfo.CurrentCulture = new CultureInfo("en-US");
+
+                RegionInfo ri = new RegionInfo(new RegionInfo(CultureInfo.CurrentCulture.Name).TwoLetterISORegionName);
+                Assert.True(RegionInfo.CurrentRegion.Equals(ri) || RegionInfo.CurrentRegion.Equals(new RegionInfo(CultureInfo.CurrentCulture.Name)));
+                Assert.Same(RegionInfo.CurrentRegion, RegionInfo.CurrentRegion);
+
+                return SuccessExitCode;
+            }).Dispose();
```
Comment (stephentoub) [inline src/System.Globalization/tests/RegionInfo/RegionInfoTests.Properties.cs]: Does this need to be in a separate process?  Or could it just use a try/catch to revert the change to the current culture?

**Comment by stephentoub** [comment]
One question, otherwise LGTM.

File: src/System.Globalization/tests/RegionInfo/RegionInfoTests.Properties.cs (hunk 1 above)
Comment (ellismg) [inline src/System.Globalization/tests/RegionInfo/RegionInfoTests.Properties.cs]: Yeah, I can do that.  I miss-remembered CurrentCulture as a process wide thing instead of thread wide and was worried that I was going to be interacting poorly with other tests in racy cases.

**Comment by ellismg** [comment]
@stephentoub Thanks, Steve, rebased with feedback applied.

**Comment by hughbe** [comment]
LGTM

**Comment by hughbe** [comment]
... But I am surprised we didn't have other tests before that verified the weird collation rules on ICU (e.g. A != a when doing an ignore case comparison)
Perhaps we don't want to barcode tests or something like that

**Comment by stephentoub** [comment]
> But I am surprised we didn't have other tests before that verified the weird collation rules on ICU (e.g. A and a are treated as different letters when doing ignore case)

I believe there are such tests.  But our CI system right now sets LANG if it's not set, so our tests in CI don't run with en-US-POSIX.

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please
Test Innerloop OSX Release Build and Test please

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please
Test Innerloop OSX Release Build and Test please

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test please
Test Innerloop OSX Release Build and Test please
